Frame.cc

```diff
@@ -889,6 +890,10 @@ bool Frame::FillSpectralProfileData(
         if (!region->IsValid()) {
             return false;
         }
+
+        // Increase the spectral profile count
+        region->IncreaseZProfileCount();
```

Comment:
There are cases where this function returns without decreasing the zprofile count (no profiles or region outside image), so the region's count will never reach zero.  Why doesn't the region inc/dec its own count when its zprofile functions are called?

Reply:
Not sure is it a good solution. Now I synchronized the zprofile counts for the Frame and the Region objects in the commit 716337aeec2a65e711e07f6543bf34194adc54f8. To make sure the zprofile process is completely done, and then destroy the objects.